### DISCUSSION TITLE
Testing : Download OSM Data in Tasking Manger Projects

### DISCUSSION THREAD

**Discussion by kshitijrajsharma** [hotosm/tasking-manager#6199] [Testing]
# Download Service in Tasking Manager

Hi everyone , We're Happy to announce that the first version of feature discussed in mid-August 2023 (reference [here](https://github.com/hotosm/tasking-manager/discussions/6037)) is now ready on the staging URL! We invite you to participate in testing and provide valuable feedback.

## About the feature : 
Our new download service enables you to easily download OSM Data within the tasking manager project boundary. The exports are tailored based on the mapping type specified in the tasking manager project. For example, if a TM Project is designated with a mapping type of `Waterways` you'll find waterways available in the download section.

Please note that the `Other` mapping type is not included in this initial phase of development. Available data will be updated **daily** if there is any activity on the project (We call it **active projects** ) . Geojson , Shapefile and KML format is available for each ! 

### Step 1: Visit the Tasking Manager staging URL

Click Here  https://tasks-stage.hotosm.org/

### Step 2: Perform the following tasks:

-   Navigate to the "Projects" section, Pick any tasking manager projects in your region 

-   Locate the "Download OSM Data" section. (It is right below contributors section and above timeline)

![image](https://github.com/hotosm/tasking-manager/assets/36752999/176709cb-dfff-4a3e-a545-2967d0007e86)

-  Choose to download outputs for any feature type (It's recommended to test all if possible).

-   Explore the data, load it into your favorite editor, navigate through the attributes and area.

-   Evaluate whether it simplifies your mapping journey , either it has any gaps or information overshared

### Step 3: Share your feedback
 
Collect your initial thoughts about this feature and share your feedback, findings or any bugs with screenshots by participating in this discussion by  **Monday 22 January.**

Thank you for your engagement, and we look forward to your valuable input!

**Comment by petya-kangalova** [comment]
Testing in HOT Tasking Manager Meet Up: https://tasks-stage.hotosm.org/projects/14455
Feedback below:

1. Yes, we can see roads and buildings, all good.
2. Author is missing- Answer: that is expected as we have no user data. 
3.  What is "source"?  It will download whatever the value was added in the source key-  Microsoft footprint, field survey, etc.
4.  Clipping boundary - that is project boundary, which you can download.  Only recent export will have the clipping boundary. It looks great!
![image](https://github.com/hotosm/tasking-manager/assets/98902727/80eec874-cd73-40f8-982f-d8ba575c930b)
5. Is the data able to be downloaded at any stage of the project, or does it only show up at a certain point?  Answer: all project data is available. If a project is updated, then the download gets updated. We run a script to update data every day!
6.  It seems like one needs to be logged into the TM to be able to download, is that correct?  Answer: yes, you do not need to be logged in to download the data.
7. Ensure that there is clarification with 'i' and in the readme file to explain that it only captures last update in tasking manager @kshitijrajsharma

**Comment by Notoncebut2x** [comment]
What a great new feature!  Thank you so much for this enhancement.  One question, does the file download only features created through the project's change sets or does it download all the features in the AoI regardless of how they were created?

**Reply by kshitijrajsharma** [reply]
Hi @Notoncebut2x  Thanks for your query ! It downloads all the features in the AOI and divides the downloads based on the mapping type defined in project !
